Prolangle/Services/MetadatumComparisonService.cs

```diff
@@ -0,0 +1,143 @@
+using Prolangle.Languages.Framework;
+using Prolangle.Models;
+using MatchType = Prolangle.Models.MatchType;
+
+namespace Prolangle.Services;
+
+public class MetadatumComparisonService(ILanguage targetLanguage)
+{
+	public TypeSystem TypingOverlap(ILanguage language)
```

Comment:
Nit: these should perhaps be `otherLanguage` for clarity.

Reply:
Agreed